ota-plus-web/app/com/advancedtelematic/auth/garage/LoginAction.scala

```diff
@@ -22,6 +25,39 @@ class LoginAction @Inject()(conf: Configuration,
   }
 }
 
+class LogoutAction @Inject()(conf: Configuration,
+                             ws: WSClient,
+                             authAction: UiAuthAction,
+                             val parser: BodyParsers.Default,
+                             val executionContext: ExecutionContext)
+  extends com.advancedtelematic.auth.LogoutAction {
+
+  private[this] val log = Logger(this.getClass)
+  private[this] val authPlusConfig = AuthPlusConfig(conf).get
+
+  override def apply(request: Request[AnyContent]): Future[Result] = {
+    AuthenticatedRequest.fromRequest(request).map(revoke)
+    Future.successful(
+      Results.Redirect(com.advancedtelematic.controllers.routes.LoginController.login()).withNewSession)
+  }
```

Comment:
I think this should be something like 

```scala
Future.fromTry(AuthenticatedRequest.fromRequest(request).map(revoke)).map { _ =>
      Results.Redirect(com.advancedtelematic.controllers.routes.LoginController.login()).withNewSession
    }
```

Otherwise the request will always succeed, even if the actual logout fails.

If we don't care about that, we should still chain the resulting Future into the Try, otherwise the request might return before `revoke` finishes.